### DISCUSSION TITLE
Prometheus exporter for a console application

### DISCUSSION THREAD

**Discussion by Deniz** [open-telemetry/opentelemetry-dotnet#4108] [Q&A] [answered]
Hi,

We have been using `OpenTelemetry.Exporter.Prometheus.AspNetCore` for our ASP .NET Core apps, however I wonder if it's possible to provide an endpoint for the grafana-agent to scrape similar to the one for  `OpenTelemetry.Exporter.Prometheus.AspNetCore`. There is also `OpenTelemetry.Exporter.Prometheus.HttpListener` in the repo though warning says it's - will - not be production ready. Is there any other option here or should just use `OpenTelemetry.Exporter.Prometheus.AspNetCore` by converting console to an ASP .NET Core one?

Thanks,

**Comment by cijothomas** [comment, marked as answer]
You are pretty much limited to this. Another alternative is use OTLP Exporter, and have it expose metrics for Prometheus for scrape.

**Reply by Deniz** [reply]
Thanks @cijothomas , to clarify are you referring the scenario in which `otel-collector` is involved in order to use OLTP Exporter? I am using grafana-agent to send the metrics to Grafana cloud, so flow is like the one below?

```
application (OLTP Exporter) -> OTel collector -> Grafana/Prometheus (via grafana-agent by using the scrape endpoint on collector)
```

OTel-collector will expose an HTTP endpoint on the collector to be scraped by Grafana Agent? Otherwise OLTP support is still in Beta stage (referring the doc [here](https://grafana.com/docs/grafana-cloud/data-configuration/otlp/send-data-otlp/)) and still needs the collector.

**Reply by cijothomas** [reply]
Yes that is right understanding of my alternative suggestion to use OTLP.
